src/components/Login/Login.tsx

```diff
@@ -33,95 +19,99 @@ export const Login = () => {
 
   const { t } = useTranslation();
 
-  const initialValues: ILogin = {
-    email: "",
-    password: "",
-  };
-
   const validationSchema = Yup.object({
     email: Yup.string()
       .email(t`user.email.validation.format`)
       .required(t`user.email.validation.requied`),
     password: Yup.string().required(t`user.password.validation.requied`),
   });
 
-  const onSubmit = async (formValue: { email: string; password: string }) => {
-    const { email, password } = formValue;
-
-    await login(email, password).then(
-      () => {
-        //Navigate to user profile
-        toast.success(t`toast.login.success`);
-      },
-      (error) => {
-        const resMessage = (error.response && error.response.data) || error.message || error.toString();
-        switch (resMessage) {
-          case '"password" length must be at least 8 characters long':
-            return toast.error(t`toast.login.length`);
-          default:
-            return toast.error(t`toast.login.error`);
-        }
-      },
-    );
+  const initialValues: ILogin = {
+    email: "",
+    password: "",
   };
 
   return (
-    <Formik initialValues={initialValues} validationSchema={validationSchema} onSubmit={onSubmit}>
-      {(formValue) => {
-        return (
-          <Form>
+    <>
+      <Formik
+        validationSchema={validationSchema}
+        initialValues={initialValues}
+        onSubmit={(values) => {
+          login(values).then(
+            () => {
+              toast.success(t`toast.login.success`);
+            },
+            (error) => {
+              const resMessage = (error.response && error.response.data) || error.message || error.toString();
+              switch (resMessage) {
+                case '"password" length must be at least 8 characters long':
+                  return toast.error(t`toast.login.length`);
+                default:
+                  return toast.error(t`toast.login.error`);
```

Comment:
jak wyżej, użyj kodu błędu jaki zwraca api a nie wiadomości

Reply:
jw. nie znam innego sposobu na zmianę wyświetlania języków pochodzących z komunikatów z serwera, ale chętnie się dowiem

Reply:
używasz error.message, spróbuj error.code, zrób consol log jak wygląda ten error, na pewno zwraca też kod błędu, nie tylko wiadomość

Reply:
w sumie racja lepiej po kodzie błędu będzie, Dzięki

Reply:
Jeszcze pytanie jak mam (StatusCodes.BAD_REQUEST) a walidacja jest z yupa to w zależnosci co wpiszę w tych polach będzie inny komunikat, tak? bo właśnie myślę jak to obsłużyć

Reply:
pod StatusCodes.BAD_REQUEST kryje się liczba 400

Reply:
const resMessage = (error.response && error.response.data) || error.message || error.toString();

zamiast error.message daj error.code (chyba)

i wystarczy tutaj if

if  (error.code === 400)  return toast.error(t`toast.login.length`);
else toast.error(t`toast.login.error`);

Reply:
a jak byś chciała to tak super uprościć to całą funkcję można spróbować zapisać tak (nie wiem czy na 100% zadziała)

```js
({code}) => toast.error((error.code === 400) ? t`toast.login.length` : t`toast.login.error`);
```

Reply:
> Jeszcze pytanie jak mam (StatusCodes.BAD_REQUEST) a walidacja jest z yupa to w zależnosci co wpiszę w tych polach będzie inny komunikat, tak? bo właśnie myślę jak to obsłużyć

ok, nie ogarnąłem pytania, Marta chyba dodawała komunikaty do Yup, zobacz jak tam to wygląda

Reply:
W zależności co wpiszę z serwera mam 400 inny kod błędu
<img width="910" alt="Screen Shot 2022-04-12 at 11 49 13 AM" src="https://user-images.githubusercontent.com/30056795/162933097-b977a405-33a7-4630-ac77-64299306c878.png">

<img width="855" alt="Screen Shot 2022-04-12 at 11 47 58 AM" src="https://user-images.githubusercontent.com/30056795/162933114-dcffff21-d961-4634-9bef-a0839100f40b.png">

@TenGosc007 może jednak zostawić tego swicha po tekście i próbować wyłapać wszystkie błędy

Reply:
![obraz](https://user-images.githubusercontent.com/56917808/162939874-549509d2-b20b-4231-a73c-4086590c1e79.png)

Reply:
wina backendu

Reply:
wszędzie jest BAD_REQUEST

Reply:
trzeba to poprawić